src/pages/guides/tutorials/stats-addon.md

```diff
@@ -0,0 +1,1093 @@
+await sandboxProxy.drawShape();
+```
+
+#### Document Sandbox
+
+```js
+// Document Sandbox
+const drawRect    = () => { /* ... */ };
+const drawEllipse = () => { /* ... */ };
+
+let _counter = 0; 👈 // private member (mind the underscore)
+
+runtime.exposeApi({
+  drawShape() { 
+    if (_counter >= 10) { // 👈 mind the underscore
+      throw new Error("Shape's budget depleted!");
+    }
+    // Randomly draw either a rectangle or an ellipse
+    Math.random() < 0.5 ? drawRect() : drawEllipse();
+	this.counter = this.counter + 1; // Use the setter to increment 
+	return this.counter; // Return the new value after incrementing
+  },
+  get counter()    { return _counter; }, // 👈 getter
+  set counter(val) { _counter = val; }   // 👈 setter
+});
+```
+
+The `counter` setter has the ability to perform additional actions, such as manipulating the document, in addition to assigning a new value to the private variable `_counter`.
+
+### Asynchronous communication
+
+The Communication API wraps all inter-context function invocations with a **Promise**. In other words, regardless of the nature of the function called, when it gets from one context to the other (from the iframe to the Document Sandbox or vice-versa), it must be dealt with as if it were **asynchronous**.
+
+```js
+// Document Sandbox
+runtime.exposeApi({
+  drawShape() { /* ... */ } // 👈 Document API methods are typically *synchronous*
+});
+
+// iframe
+❌ const res = sandboxProxy.drawShape(); // 👀 no await, returns a Promise
+❌ console.log(res instanceof Promise);  // true
+
+✅ const res = await sandboxProxy.drawShape(); // using await
+✅ console.log(typeof res); // number (it returns the counter, as you remember)
+```
+
+## Coding the Stats add-on
+
+You are now equipped with all the theory and reference snippets needed to start building the Stats add-on: it's a simple project implementing the Communication API in a couple of different ways. Feature-wise, it's split into two parts.
+
+1. The add-on shows two **status lights** that indicate whether the SDKs are ready for use.
+1. At the press of a button, the add-on (via Document API) collects the document's metadata[^0], which is used to compile and show a **table of Nodes** (elements).
+
+![](images/stats-addon-vscode.png)
+
+Like in the [Grids add-on tutorial](grids-addon.md), the starting point will be [this template](https://github.com/AdobeDocs/express-add-on-samples/tree/main/script-runtime-samples/express-addon-document-api-template), which provides a Webpack-managed JavaScript—hence, able to easily import Spectrum Web Components to build the User Interface. Everything's in the `src` folder:
+
+- `index.html` is where the iframe's UI is built.
+- `ui/index.js` deals with the add-on's internal logic.
+- `ui/table-utils.js` contains table-related functions consumed by the iframe and imported by `index.js` to keep it slim.
+- `documentSandbox/code.js` contains the Document API methods exposed to the iframe.
```

Comment:
nit: the folder name can be just `sandbox` ? `documentSandbox` might be bit long.

Reply:
I did consult with Kerri and Peter when the "Big Rename" happened—this is what they suggested using, possibly as a way to future-proof it if/when new sandboxes will be released.